dags/synaptor_dags.py

```diff
@@ -0,0 +1,111 @@
+"""DAG definition for synaptor workflows."""
+from datetime import datetime
+from configparser import ConfigParser
+
+from airflow import DAG
+from airflow.models import Variable
+
+from helper_ops import scale_up_cluster_op, scale_down_cluster_op
+from param_default import default_synaptor_param
+from synaptor_ops import manager_op, drain_op
+from synaptor_ops import synaptor_op, wait_op, generate_op
+
+
+# Processing parameters
+# We need to set a default so any airflow container can parse the dag before we
+# pass in a configuration file
+
+try:
+    param = Variable.get("synaptor_param", default_synaptor_param)
+    cp = ConfigParser()
+    cp.read_string(param)
+    MAX_CLUSTER_SIZE = cp.getint("Workflow", "maxclustersize")
+
+except:  # not sure why this doesn't work sometimes
+    MAX_CLUSTER_SIZE = 1
+
```

Comment:
The bot stores the information about the clusters in the "InstanceGroups" connection, you can use that to decide the number of instance the task should use.

Reply:
Ok. I'll make this change later as well.